# Aviation Accidents Analysis

## 1. Overview

A company I am working with is planning to diversify its portfolio by exploring new areas, particularly the aviation industry. The leadership is interested in the risks associated with aircraft operations, mainly safety and reliability. It is on that background that as a senior data scientist in the company that the management tusked me with the responsibility of carrying out data analysis to assist the management in making informed investment decisions to determine which aircraft presents the lowest operational risk.


## 2. Business Understanding
To find out the safest and most reliable aircraft for the company to purchase based on an analysis of accident and incident data collected over the years. Therefore, the company wants to:
* Minimize accident frequency
* low fetality count


## 3. Data Science Understanding
Analyze accident data collected over the years to establish which aircraft has the lowest risk by using risk indicators such as accident frequency, fatality rate, and severity of injuries.
This will involve the following analysis:
* Data exploration
* Data Cleaning
* Grouping aircraft by make and model
* Calculating the safety matrix per aircraft type
* Visualizing patterns of high-risk and low-risk aircrafts


In [1]:
# importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

Intel MKL WARNING: Support of Intel(R) Streaming SIMD Extensions 4.2 (Intel(R) SSE4.2) enabled only processors has been deprecated. Intel oneAPI Math Kernel Library 2025.0 will require Intel(R) Advanced Vector Extensions (Intel(R) AVX) instructions.
Intel MKL WARNING: Support of Intel(R) Streaming SIMD Extensions 4.2 (Intel(R) SSE4.2) enabled only processors has been deprecated. Intel oneAPI Math Kernel Library 2025.0 will require Intel(R) Advanced Vector Extensions (Intel(R) AVX) instructions.


In [2]:
# Loading accident data to the notebook
aviation_data = pd.read_csv("../Data/AviationData.csv", index_col = 0, encoding = "latin1", low_memory = False)

### 3.1 exploring data

In [3]:
aviation_data.head() # checking the first five rows of the dataframe

Investigation.Type Accident.Number  Event.Date  \
Event.Id                                                        
20001218X45444           Accident      SEA87LA080  1948-10-24   
20001218X45447           Accident      LAX94LA336  1962-07-19   
20061025X01555           Accident      NYC07LA005  1974-08-30   
20001218X45448           Accident      LAX96LA321  1977-06-19   
20041105X01764           Accident      CHI79FA064  1979-08-02   

                       Location        Country   Latitude   Longitude  \
Event.Id                                                                
20001218X45444  MOOSE CREEK, ID  United States        NaN         NaN   
20001218X45447   BRIDGEPORT, CA  United States        NaN         NaN   
20061025X01555    Saltville, VA  United States  36.922223  -81.878056   
20001218X45448       EUREKA, CA  United States        NaN         NaN   
20041105X01764       Canton, OH  United States        NaN         NaN   

               Airport.Code Airport.Name Injury.Severity  ...  \
Event.Id                                                  ...   
20001218X45444          NaN          NaN        Fatal(2)  ...   
20001218X45447          NaN          NaN        Fatal(4)  ...   
20061025X01555          NaN          NaN        Fatal(3)  ...   
20001218X45448          NaN          NaN        Fatal(2)  ...   
20041105X01764          NaN          NaN        Fatal(1)  ...   

               Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
Event.Id                                                            
20001218X45444          Personal         NaN                  2.0   
20001218X45447          Personal         NaN                  4.0   
20061025X01555          Personal         NaN                  3.0   
20001218X45448          Personal         NaN                  2.0   
20041105X01764          Personal         NaN                  1.0   

               Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
Event.Id                                                                     
20001218X45444                    0.0                  0.0             0.0   
20001218X45447                    0.0                  0.0             0.0   
20061025X01555                    NaN                  NaN             NaN   
20001218X45448                    0.0                  0.0             0.0   
20041105X01764                    2.0                  NaN             0.0   

                Weather.Condition Broad.phase.of.flight   Report.Status  \
Event.Id                                                                  
20001218X45444                UNK                Cruise  Probable Cause   
20001218X45447                UNK               Unknown  Probable Cause   
20061025X01555                IMC                Cruise  Probable Cause   
20001218X45448                IMC                Cruise  Probable Cause   
20041105X01764                VMC              Approach  Probable Cause   

               Publication.Date  
Event.Id                         
20001218X45444              NaN  
20001218X45447       19-09-1996  
20061025X01555       26-02-2007  
20001218X45448       12-09-2000  
20041105X01764       16-04-1980  

[5 rows x 30 columns]

In [4]:
aviation_data.tail() # chwcking the last five rows of the dataframe

Investigation.Type Accident.Number  Event.Date       Location  \
Event.Id                                                                       
20221227106491           Accident      ERA23LA093  2022-12-26  Annapolis, MD   
20221227106494           Accident      ERA23LA095  2022-12-26    Hampton, NH   
20221227106497           Accident      WPR23LA075  2022-12-26     Payson, AZ   
20221227106498           Accident      WPR23LA076  2022-12-26     Morgan, UT   
20221230106513           Accident      ERA23LA097  2022-12-29     Athens, GA   

                      Country Latitude Longitude Airport.Code Airport.Name  \
Event.Id                                                                     
20221227106491  United States      NaN       NaN          NaN          NaN   
20221227106494  United States      NaN       NaN          NaN          NaN   
20221227106497  United States  341525N  1112021W          PAN       PAYSON   
20221227106498  United States      NaN       NaN          NaN          NaN   
20221230106513  United States      NaN       NaN          NaN          NaN   

               Injury.Severity  ... Purpose.of.flight         Air.carrier  \
Event.Id                        ...                                         
20221227106491           Minor  ...          Personal                 NaN   
20221227106494             NaN  ...               NaN                 NaN   
20221227106497       Non-Fatal  ...          Personal                 NaN   
20221227106498             NaN  ...          Personal  MC CESSNA 210N LLC   
20221230106513           Minor  ...          Personal                 NaN   

               Total.Fatal.Injuries Total.Serious.Injuries  \
Event.Id                                                     
20221227106491                  0.0                    1.0   
20221227106494                  0.0                    0.0   
20221227106497                  0.0                    0.0   
20221227106498                  0.0                    0.0   
20221230106513                  0.0                    1.0   

               Total.Minor.Injuries Total.Uninjured  Weather.Condition  \
Event.Id                                                                 
20221227106491                  0.0             0.0                NaN   
20221227106494                  0.0             0.0                NaN   
20221227106497                  0.0             1.0                VMC   
20221227106498                  0.0             0.0                NaN   
20221230106513                  0.0             1.0                NaN   

               Broad.phase.of.flight Report.Status Publication.Date  
Event.Id                                                             
20221227106491                   NaN           NaN       29-12-2022  
20221227106494                   NaN           NaN              NaN  
20221227106497                   NaN           NaN       27-12-2022  
20221227106498                   NaN           NaN              NaN  
20221230106513                   NaN           NaN       30-12-2022  

[5 rows x 30 columns]

In [5]:
aviation_data.info() # getting concise information about the dataframe

<class 'pandas.core.frame.DataFrame'>
Index: 88889 entries, 20001218X45444 to 20221230106513
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Investigation.Type      88889 non-null  object 
 1   Accident.Number         88889 non-null  object 
 2   Event.Date              88889 non-null  object 
 3   Location                88837 non-null  object 
 4   Country                 88663 non-null  object 
 5   Latitude                34382 non-null  object 
 6   Longitude               34373 non-null  object 
 7   Airport.Code            50132 non-null  object 
 8   Airport.Name            52704 non-null  object 
 9   Injury.Severity         87889 non-null  object 
 10  Aircraft.damage         85695 non-null  object 
 11  Aircraft.Category       32287 non-null  object 
 12  Registration.Number     87507 non-null  object 
 13  Make                    88826 non-null  object 
 14  Model                

All the columns have missing values except the first three.

In [6]:
aviation_data.shape # checking the dataframe dimensions, i.e number of rows and cloumns in the dataframe.

(88889, 30)

Dataset contains 88,889 rows and 30 columns.

## 4. Data Preparation

### 4.1 Checking for missing values

In [7]:
aviation_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 88889 entries, 20001218X45444 to 20221230106513
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Investigation.Type      88889 non-null  object 
 1   Accident.Number         88889 non-null  object 
 2   Event.Date              88889 non-null  object 
 3   Location                88837 non-null  object 
 4   Country                 88663 non-null  object 
 5   Latitude                34382 non-null  object 
 6   Longitude               34373 non-null  object 
 7   Airport.Code            50132 non-null  object 
 8   Airport.Name            52704 non-null  object 
 9   Injury.Severity         87889 non-null  object 
 10  Aircraft.damage         85695 non-null  object 
 11  Aircraft.Category       32287 non-null  object 
 12  Registration.Number     87507 non-null  object 
 13  Make                    88826 non-null  object 
 14  Model                

In [8]:
missing_values_percentage= aviation_data.isnull().mean().sort_values(ascending = False) * 100
missing_values_percentage

Schedule                  85.845268
Air.carrier               81.271023
FAR.Description           63.974170
Aircraft.Category         63.677170
Longitude                 61.330423
Latitude                  61.320298
Airport.Code              43.601570
Airport.Name              40.708074
Broad.phase.of.flight     30.560587
Publication.Date          15.492356
Total.Serious.Injuries    14.073732
Total.Minor.Injuries      13.424608
Total.Fatal.Injuries      12.826109
Engine.Type                7.982990
Report.Status              7.181991
Purpose.of.flight          6.965991
Number.of.Engines          6.844491
Total.Uninjured            6.650992
Weather.Condition          5.053494
Aircraft.damage            3.593246
Registration.Number        1.554748
Injury.Severity            1.124999
Country                    0.254250
Amateur.Built              0.114750
Model                      0.103500
Make                       0.070875
Location                   0.058500
Accident.Number            0

In [9]:
# columns with more than 60 percent missing values
columns_to_drop = missing_values_percentage.head(6) 
columns_to_drop

Schedule             85.845268
Air.carrier          81.271023
FAR.Description      63.974170
Aircraft.Category    63.677170
Longitude            61.330423
Latitude             61.320298
dtype: float64

In [10]:
# Dropping columns that have more than 60 percent missing values
aviation_data = aviation_data.drop(columns= columns_to_drop.index)
aviation_data

Investigation.Type Accident.Number  Event.Date  \
Event.Id                                                        
20001218X45444           Accident      SEA87LA080  1948-10-24   
20001218X45447           Accident      LAX94LA336  1962-07-19   
20061025X01555           Accident      NYC07LA005  1974-08-30   
20001218X45448           Accident      LAX96LA321  1977-06-19   
20041105X01764           Accident      CHI79FA064  1979-08-02   
...                           ...             ...         ...   
20221227106491           Accident      ERA23LA093  2022-12-26   
20221227106494           Accident      ERA23LA095  2022-12-26   
20221227106497           Accident      WPR23LA075  2022-12-26   
20221227106498           Accident      WPR23LA076  2022-12-26   
20221230106513           Accident      ERA23LA097  2022-12-29   

                       Location        Country Airport.Code Airport.Name  \
Event.Id                                                                   
20001218X45444  MOOSE CREEK, ID  United States          NaN          NaN   
20001218X45447   BRIDGEPORT, CA  United States          NaN          NaN   
20061025X01555    Saltville, VA  United States          NaN          NaN   
20001218X45448       EUREKA, CA  United States          NaN          NaN   
20041105X01764       Canton, OH  United States          NaN          NaN   
...                         ...            ...          ...          ...   
20221227106491    Annapolis, MD  United States          NaN          NaN   
20221227106494      Hampton, NH  United States          NaN          NaN   
20221227106497       Payson, AZ  United States          PAN       PAYSON   
20221227106498       Morgan, UT  United States          NaN          NaN   
20221230106513       Athens, GA  United States          NaN          NaN   

               Injury.Severity Aircraft.damage Registration.Number  ...  \
Event.Id                                                            ...   
20001218X45444        Fatal(2)       Destroyed              NC6404  ...   
20001218X45447        Fatal(4)       Destroyed              N5069P  ...   
20061025X01555        Fatal(3)       Destroyed              N5142R  ...   
20001218X45448        Fatal(2)       Destroyed              N1168J  ...   
20041105X01764        Fatal(1)       Destroyed               N15NY  ...   
...                        ...             ...                 ...  ...   
20221227106491           Minor             NaN              N1867H  ...   
20221227106494             NaN             NaN              N2895Z  ...   
20221227106497       Non-Fatal     Substantial              N749PJ  ...   
20221227106498             NaN             NaN              N210CU  ...   
20221230106513           Minor             NaN              N9026P  ...   

                  Engine.Type Purpose.of.flight Total.Fatal.Injuries  \
Event.Id                                                               
20001218X45444  Reciprocating          Personal                  2.0   
20001218X45447  Reciprocating          Personal                  4.0   
20061025X01555  Reciprocating          Personal                  3.0   
20001218X45448  Reciprocating          Personal                  2.0   
20041105X01764            NaN          Personal                  1.0   
...                       ...               ...                  ...   
20221227106491            NaN          Personal                  0.0   
20221227106494            NaN               NaN                  0.0   
20221227106497            NaN          Personal                  0.0   
20221227106498            NaN          Personal                  0.0   
20221230106513            NaN          Personal                  0.0   

                Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
Event.Id                                                                      
20001218X45444                     0.0                  0.0             0.0   
20001218X45447                

In [11]:
aviation_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 88889 entries, 20001218X45444 to 20221230106513
Data columns (total 24 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Investigation.Type      88889 non-null  object 
 1   Accident.Number         88889 non-null  object 
 2   Event.Date              88889 non-null  object 
 3   Location                88837 non-null  object 
 4   Country                 88663 non-null  object 
 5   Airport.Code            50132 non-null  object 
 6   Airport.Name            52704 non-null  object 
 7   Injury.Severity         87889 non-null  object 
 8   Aircraft.damage         85695 non-null  object 
 9   Registration.Number     87507 non-null  object 
 10  Make                    88826 non-null  object 
 11  Model                   88797 non-null  object 
 12  Amateur.Built           88787 non-null  object 
 13  Number.of.Engines       82805 non-null  float64
 14  Engine.Type          

### 4.2 Filling the missing values

In [12]:
categorical_columns = ["Investigation.Type", "Accident.Number", "Event.Date", "Location", "Country", "Airport.Code", "Airport.Name", "Injury.Severity", "Aircraft.damage", "Registration.Number", "Make", "Model", "Amateur.Built", "Purpose.of.flight", "Weather.Condition", "Broad.phase.of.flight", "Report.Status","Publication.Date"]
categorical_columns

['Investigation.Type',
 'Accident.Number',
 'Event.Date',
 'Location',
 'Country',
 'Airport.Code',
 'Airport.Name',
 'Injury.Severity',
 'Aircraft.damage',
 'Registration.Number',
 'Make',
 'Model',
 'Amateur.Built',
 'Purpose.of.flight',
 'Weather.Condition',
 'Broad.phase.of.flight',
 'Report.Status',
 'Publication.Date']

In [17]:
aviation_data[categorical_columns] = aviation_data[categorical_columns].fillna("unknown")
aviation_data

Investigation.Type Accident.Number  Event.Date  \
Event.Id                                                        
20001218X45444           Accident      SEA87LA080  1948-10-24   
20001218X45447           Accident      LAX94LA336  1962-07-19   
20061025X01555           Accident      NYC07LA005  1974-08-30   
20001218X45448           Accident      LAX96LA321  1977-06-19   
20041105X01764           Accident      CHI79FA064  1979-08-02   
...                           ...             ...         ...   
20221227106491           Accident      ERA23LA093  2022-12-26   
20221227106494           Accident      ERA23LA095  2022-12-26   
20221227106497           Accident      WPR23LA075  2022-12-26   
20221227106498           Accident      WPR23LA076  2022-12-26   
20221230106513           Accident      ERA23LA097  2022-12-29   

                       Location        Country Airport.Code Airport.Name  \
Event.Id                                                                   
20001218X45444  MOOSE CREEK, ID  United States      unknown      unknown   
20001218X45447   BRIDGEPORT, CA  United States      unknown      unknown   
20061025X01555    Saltville, VA  United States      unknown      unknown   
20001218X45448       EUREKA, CA  United States      unknown      unknown   
20041105X01764       Canton, OH  United States      unknown      unknown   
...                         ...            ...          ...          ...   
20221227106491    Annapolis, MD  United States      unknown      unknown   
20221227106494      Hampton, NH  United States      unknown      unknown   
20221227106497       Payson, AZ  United States          PAN       PAYSON   
20221227106498       Morgan, UT  United States      unknown      unknown   
20221230106513       Athens, GA  United States      unknown      unknown   

               Injury.Severity Aircraft.damage Registration.Number  ...  \
Event.Id                                                            ...   
20001218X45444        Fatal(2)       Destroyed              NC6404  ...   
20001218X45447        Fatal(4)       Destroyed              N5069P  ...   
20061025X01555        Fatal(3)       Destroyed              N5142R  ...   
20001218X45448        Fatal(2)       Destroyed              N1168J  ...   
20041105X01764        Fatal(1)       Destroyed               N15NY  ...   
...                        ...             ...                 ...  ...   
20221227106491           Minor         unknown              N1867H  ...   
20221227106494         unknown         unknown              N2895Z  ...   
20221227106497       Non-Fatal     Substantial              N749PJ  ...   
20221227106498         unknown         unknown              N210CU  ...   
20221230106513           Minor         unknown              N9026P  ...   

                  Engine.Type Purpose.of.flight Total.Fatal.Injuries  \
Event.Id                                                               
20001218X45444  Reciprocating          Personal                  2.0   
20001218X45447  Reciprocating          Personal                  4.0   
20061025X01555  Reciprocating          Personal                  3.0   
20001218X45448  Reciprocating          Personal                  2.0   
20041105X01764            NaN          Personal                  1.0   
...                       ...               ...                  ...   
20221227106491            NaN          Personal                  0.0   
20221227106494            NaN           unknown                  0.0   
20221227106497            NaN          Personal                  0.0   
20221227106498            NaN          Personal                  0.0   
20221230106513            NaN          Personal                  0.0   

                Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
Event.Id                                                                      
20001218X45444                     0.0                  0.0             0.0   
20001218X45447                

In [18]:
aviation_data.sample(10)

Investigation.Type Accident.Number  Event.Date  \
Event.Id                                                        
20080304X00254           Incident      ENG08RA015  2008-01-24   
20020613X00875           Accident      LAX02LA186  2002-06-07   
20080811X01206           Accident      DFW08WA201  2008-08-02   
20020917X03715           Accident      LAX82FVD20  1982-08-04   
20001208X07157           Accident      FTW97LA058  1996-12-09   
20001213X29196           Accident      MIA89LA232  1989-08-25   
20001213X29292           Accident      ATL89FA214  1989-09-17   
20100513X45441           Accident      CEN09WA617  2009-05-31   
20001211X14470           Accident      LAX92T#A04  1992-04-20   
20120503X01852           Accident      CEN12LA267  2012-05-02   

                           Location        Country Airport.Code  \
Event.Id                                                          
20080304X00254    Kingston, Jamaica        Jamaica      unknown   
20020613X00875        HENDERSON, NV  United States          HND   
20080811X01206  Guadalajara, Mexico         Mexico         MMGL   
20020917X03715        LIVERMORE, CA  United States      unknown   
20001208X07157      EAST CAMERON 71        unknown      unknown   
20001213X29196          EDWARDS, MS  United States      unknown   
20001213X29292           GORDON, AL  United States      unknown   
20100513X45441   Strovikion, Greece         Greece      unknown   
20001211X14470       FORT IRWIN, CA  United States      unknown   
20120503X01852     Valley Falls, KS  United States      unknown   

                            Airport.Name Injury.Severity Aircraft.damage  \
Event.Id                                                                   
20080304X00254                   unknown         unknown         unknown   
20020613X00875                 Henderson       Non-Fatal     Substantial   
20080811X01206  Guadalajara Intl Airport       Non-Fatal       Destroyed   
20020917X03715                   unknown       Non-Fatal     Substantial   
20001208X07157                   unknown       Non-Fatal     Substantial   
20001213X29196                   unknown       Non-Fatal     Substantial   
20001213X29292                   unknown        Fatal(3)     Substantial   
20100513X45441         Airfield Kopaidas           Fatal     Substantial   
20001211X14470                   unknown       Non-Fatal     Substantial   
20120503X01852                   unknown       Non-Fatal     Substantial   

               Registration.Number  ...    Engine.Type   Purpose.of.flight  \
Event.Id                            ...                                      
20080304X00254              6Y-JRG  ...            NaN             unknown   
20020613X00875              N49672  ...  Reciprocating       Instructional   
20080811X01206              XB-KPB  ...            NaN             Unknown   
20020917X03715              N6976N  ...  Reciprocating            Business   
20001208X07157              N390MA  ...    Turbo Shaft             Unknown   
20001213X29196              N53327  ...  Reciprocating  Aerial Application   
20001213X29292              N7503J  ...  Reciprocating            Personal   
20100513X45441            OK-JUG19  ...  Reciprocating             unknown   
20001211X14470              N5736N  ...        Unknown     Public Aircraft   
20120503X01852               N85RB  ...    Turbo Shaft  Aerial Observation   

               Total.Fatal.Injuries  Total.Serious.Injuries  \
Event.Id                                                      
20080304X00254                  0.0                     0.0   
20020613X00875                  0.0                     0.0   
20080811X01206                  0.0                     0.0   
20020917X03715                  0.0                     0.0   
20001208X07157                  0.0                     0.0   
20001213X29196                  0.0                     0.0   
20001213X29292                  3.0                     0.0   
20100513X454

In [14]:
numericals_columns = ["Total.Fatal.Injuries", "Total.Serious.Injuries", "Total.Minor.Injuries", "Total.Uninjured", "Number.of.Engines"]
numericals_columns

['Total.Fatal.Injuries',
 'Total.Serious.Injuries',
 'Total.Minor.Injuries',
 'Total.Uninjured',
 'Number.of.Engines']

In [15]:
aviation_data[numericals_columns] = aviation_data[numericals_columns].fillna(0)
aviation_data

Investigation.Type Accident.Number  Event.Date  \
Event.Id                                                        
20001218X45444           Accident      SEA87LA080  1948-10-24   
20001218X45447           Accident      LAX94LA336  1962-07-19   
20061025X01555           Accident      NYC07LA005  1974-08-30   
20001218X45448           Accident      LAX96LA321  1977-06-19   
20041105X01764           Accident      CHI79FA064  1979-08-02   
...                           ...             ...         ...   
20221227106491           Accident      ERA23LA093  2022-12-26   
20221227106494           Accident      ERA23LA095  2022-12-26   
20221227106497           Accident      WPR23LA075  2022-12-26   
20221227106498           Accident      WPR23LA076  2022-12-26   
20221230106513           Accident      ERA23LA097  2022-12-29   

                       Location        Country Airport.Code Airport.Name  \
Event.Id                                                                   
20001218X45444  MOOSE CREEK, ID  United States      unknown      unknown   
20001218X45447   BRIDGEPORT, CA  United States      unknown      unknown   
20061025X01555    Saltville, VA  United States      unknown      unknown   
20001218X45448       EUREKA, CA  United States      unknown      unknown   
20041105X01764       Canton, OH  United States      unknown      unknown   
...                         ...            ...          ...          ...   
20221227106491    Annapolis, MD  United States      unknown      unknown   
20221227106494      Hampton, NH  United States      unknown      unknown   
20221227106497       Payson, AZ  United States          PAN       PAYSON   
20221227106498       Morgan, UT  United States      unknown      unknown   
20221230106513       Athens, GA  United States      unknown      unknown   

               Injury.Severity Aircraft.damage Registration.Number  ...  \
Event.Id                                                            ...   
20001218X45444        Fatal(2)       Destroyed              NC6404  ...   
20001218X45447        Fatal(4)       Destroyed              N5069P  ...   
20061025X01555        Fatal(3)       Destroyed              N5142R  ...   
20001218X45448        Fatal(2)       Destroyed              N1168J  ...   
20041105X01764        Fatal(1)       Destroyed               N15NY  ...   
...                        ...             ...                 ...  ...   
20221227106491           Minor         unknown              N1867H  ...   
20221227106494         unknown         unknown              N2895Z  ...   
20221227106497       Non-Fatal     Substantial              N749PJ  ...   
20221227106498         unknown         unknown              N210CU  ...   
20221230106513           Minor         unknown              N9026P  ...   

                  Engine.Type Purpose.of.flight Total.Fatal.Injuries  \
Event.Id                                                               
20001218X45444  Reciprocating          Personal                  2.0   
20001218X45447  Reciprocating          Personal                  4.0   
20061025X01555  Reciprocating          Personal                  3.0   
20001218X45448  Reciprocating          Personal                  2.0   
20041105X01764            NaN          Personal                  1.0   
...                       ...               ...                  ...   
20221227106491            NaN          Personal                  0.0   
20221227106494            NaN           unknown                  0.0   
20221227106497            NaN          Personal                  0.0   
20221227106498            NaN          Personal                  0.0   
20221230106513            NaN          Personal                  0.0   

                Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
Event.Id                                                                      
20001218X45444                     0.0                  0.0             0.0   
20001218X45447                

In [16]:
aviation_data.sample(10)

Investigation.Type Accident.Number  Event.Date  \
Event.Id                                                        
20001213X25058           Accident      DEN88LA077  1988-02-21   
20130911X90214           Accident      ERA13WA411  2013-09-09   
20100913X14749           Accident      ANC10LA084  2010-09-11   
20080118X00073           Accident      SEA08LA061  2008-01-14   
20001212X17733           Accident      ANC91FA128  1991-08-16   
20090518X43152           Accident      ERA09LA293  2009-05-17   
20210518103095           Accident      CEN21LA222  2021-05-18   
20030702X01010           Accident      LAX03WA222  2003-06-20   
20060707X00889           Accident      CHI06CA156  2006-06-19   
20211118104270           Accident      ERA22LA066  2021-11-12   

                          Location        Country Airport.Code  \
Event.Id                                                         
20001213X25058            GILA, NM  United States      unknown   
20130911X90214   Valparaiso, Chile          Chile      unknown   
20100913X14749    South Naknek, AK  United States      unknown   
20080118X00073   San Francisco, CA  United States         KSFO   
20001212X17733            RUBY, AK  United States      unknown   
20090518X43152          Camden, NC  United States      unknown   
20210518103095          Racine, WI  United States          RAC   
20030702X01010   Camden, Australia      Australia      unknown   
20060707X00889     COOPERSTOWN, ND  United States      unknown   
20211118104270  St. Petersburg, FL  United States      unknown   

                               Airport.Name Injury.Severity Aircraft.damage  \
Event.Id                                                                      
20001213X25058                      unknown       Non-Fatal     Substantial   
20130911X90214                 Vina Del Mar           Fatal       Destroyed   
20100913X14749                      unknown       Non-Fatal     Substantial   
20080118X00073  San Francisco International       Non-Fatal     Substantial   
20001212X17733                      unknown        Fatal(3)       Destroyed   
20090518X43152                      unknown       Non-Fatal     Substantial   
20210518103095                  BATTEN INTL       Non-Fatal       Destroyed   
20030702X01010                      unknown        Fatal(2)       Destroyed   
20060707X00889                      unknown       Non-Fatal     Substantial   
20211118104270                      unknown       Non-Fatal     Substantial   

               Registration.Number  ...    Engine.Type Purpose.of.flight  \
Event.Id                            ...                                    
20001213X25058              N2359E  ...  Reciprocating          Personal   
20130911X90214              CC-CNW  ...            NaN           unknown   
20100913X14749              N7414K  ...  Reciprocating          Personal   
20080118X00073              N705SK  ...      Turbo Fan           unknown   
20001212X17733              N9026E  ...  Reciprocating          Personal   
20090518X43152               N92RG  ...  Reciprocating          Personal   
20210518103095              N521CT  ...            NaN     Instructional   
20030702X01010             unknown  ...            NaN     Instructional   
20060707X00889              N9073G  ...  Reciprocating          Personal   
20211118104270              N5862W  ...            NaN          Personal   

               Total.Fatal.Injuries  Total.Serious.Injuries  \
Event.Id                                                      
20001213X25058                  0.0                     1.0   
20130911X90214                  2.0                     0.0   
20100913X14749                  0.0                     0.0   
20080118X00073                  0.0                     0.0   
20001212X17733                  3.0                     0.0   
20090518X43152                  0.0                     0.0   
20210518103095                  0.0                     1.0   
20030702X010

### 4.1 Checking and handling duplicated values

## 5. Exploratory Data Analysis

## 6. Recommendations